Sequence divergence in venom genes within and between montane pitvipers
(*Viperidae* :*Crotalinae*  :*Cerrophidion*) species is driven by
mutation-drift equilibrium.
================
Ramses A. Rosales-Garcia, Rhett M. Rautsaw , Erich P. Hofmann, Christoph
I. Grunwald, Jason M. Jones, Hector Franz-Chavez, Ivan T.
Ahumada-Carrillo, Ricardo Ramirez-Chaparro, Miguel Angel De la
Torre-Loranca, Jason L. Strickland, Andrew J. Mason, Matthew L. Holding,
Miguel Borja, Gamaliel Castaneda-Gaytan, Darin R. Rokyta, Tristan D.
Schramer, N. Jade Mellor, Edward A. Myers, Christopher Parkinson
2023 April 19

- <a href="#first_part" id="toc-first_part">first_part</a>
- <a href="#second_part" id="toc-second_part">second_part</a>
- <a href="#supplementary" id="toc-supplementary">Supplementary</a>
- <a href="#data" id="toc-data">Data</a>

<figure>
<img src="pictures/C_tzotzilorum1.jpg" style="width:500.0%"
alt="Cerrophidion tzotzilorum" />
<figcaption aria-hidden="true"><em>Cerrophidion
tzotzilorum</em></figcaption>
</figure>

This Github page contains the code and scripts used in the development
of the paper.

# first_part

The [first part](firts_part/Scripts_Paper.md) contains the shell code
used to assembly of transcriptomes, annotation, expression analysis,
phylogenetics, variant calling and selection analisys. There are several
[scripts](scripts/Scripts_Description.md) used that can be found in the
[scripts](scripts) folder. Some scripts are owned by a third party, in
those cases we mentioned the location of the original script or the
contact were the scripts might be found.

# second_part

The [second part](second_part/Cgodm_selection_BMC.md) contains the R
scripts with the statistics analysis that compare selection signals in
toxins vs nontoxins, the necessary data to run the analysis might be
found withing the folder.

# Supplementary

The
[supplementary](https://github.com/RamsesRosales/Cerrophidion_Selection/tree/main/supplementary)
directory contains the suplementary material submitted to the table.

- Additional_Figures.pdf
- Additional_Tables.xlsx
- Annotation_Scripts.html
- Selection_Scripts.html

# Data

The
[data](https://github.com/RamsesRosales/Cerrophidion_Selection/tree/main/data)
directory contains the [consensus
transcriptomes](https://github.com/RamsesRosales/Cerrophidion_Selection/tree/main/data/Transcriptomes),
the [RSEM
results](https://github.com/RamsesRosales/Cerrophidion_Selection/tree/main/data/Expression)
and other files generated and analised, during this work.
